File: sdk-node/examples/prices/retrieve.ts
/**
 * Retrieve a single price by ID, including its recurring cadence and feature
 * grants.
 *
 * Run:
 *   npx tsx examples/prices/retrieve.ts
 */

import { ThreeCommon } from '@3common/sdk'

const client = new ThreeCommon({
  apiKey: '3co_your_api_key_here',
})

const price = await client.prices.retrieve('price_replace_with_real_id')

console.log(`price ${price.id ?? '?'} [${price.type ?? '?'}]`)
console.log(`  product  ${price.productId ?? '?'}`)
console.log(`  amount   ${String(price.unitAmount ?? 0)} ${price.currency ?? ''}`)
if (price.recurring) {
  console.log(
    `  cadence  every ${String(price.recurring.intervalCount)} ${price.recurring.interval}`,
  )
}
for (const feature of price.features ?? []) {
  console.log(`  feature  ${feature.featureKey} [${feature.type}]`)
}

File: sdk-python/examples/prices/retrieve.py
"""Run with: python examples/prices/retrieve.py"""

from __future__ import annotations

from threecommon import ThreeCommon


def main() -> None:
    with ThreeCommon(api_key="3co_your_api_key_here") as client:
        price = client.prices.retrieve("price_replace_with_real_id")
        print(f"price {price.id} [{price.type}]")
        print(f"  product  {price.product_id}")
        print(f"  amount   {price.unit_amount} {price.currency}")
        if price.recurring is not None:
            print(f"  cadence  every {price.recurring.interval_count} {price.recurring.interval}")
        for feature in price.features or []:
            print(f"  feature  {feature.feature_key} [{feature.type}]")


if __name__ == "__main__":
    main()
